pkg/items/itemsService.go

```diff
@@ -51,33 +51,48 @@ func (ish ItemServiceHandler) ServeHTTP(w http.ResponseWriter, r *http.Request)
 			return
 		}
 	case "edit":
-		err := utils.RenderTemplate(w, nil, serviceEndpoint+"edit.html")
+		pathArray := strings.Split(strings.TrimPrefix(r.URL.Path, serviceEndpoint), "/")
+		response, err := utils.HandleGetSingleElementRequest(r, ish, getSingleItemQuery, pathArray[len(pathArray)-2])
+		if err != nil {
```

Comment:
Not really thinking of any way to resolve this, but part of me is bothered that this basic conditional shows up (with essentially no variation) multiple times.

Reply:
There's a way to remove some of redundant code but ultimately we're stuck with some of this due ot the nature of Go.